extensions/iceberg/src/main/java/io/deephaven/iceberg/util/IcebergTools.java

```diff
@@ -16,4 +22,62 @@ public static IcebergCatalogAdapter createAdapter(
             final FileIO fileIO) {
         return new IcebergCatalogAdapter(catalog, fileIO);
     }
+
+    /**
+     * <p>
+     * Create an Iceberg catalog adapter for an Iceberg catalog created from configuration properties. These properties
+     * map to the Iceberg catalog Java API properties and are used to create the catalog and file IO implementations.
+     * </p>
+     * <p>
+     * This is a wrapper around {@link CatalogUtil#buildIcebergCatalog(String, Map, Object)} and accepts the same
+     * properties. The minimal set of properties required to create an Iceberg catalog are:
+     * <ul>
+     * <li>{@link CatalogProperties#CATALOG_IMPL}</li>
+     * <li>{@link CatalogProperties#URI}</li>
+     * <li>{@link CatalogProperties#WAREHOUSE_LOCATION}</li>
+     * <li>{@link CatalogProperties#FILE_IO_IMPL}</li>
+     * </ul>
+     * <p>
+     * Additional properties for the specific catalog should also be included, such as as S3-specific properties for
+     * authentication or endpoint overriding.
+     * </p>
+     * 
+     * @param name the name of the catalog; if omitted, the catalog URI will be used to generate a name
+     * @param properties the map containing the Iceberg catalog properties to use
+     * @return the Iceberg catalog adapter
+     */
+    public static IcebergCatalogAdapter createAdapter(
+            @Nullable final String name,
+            @NotNull final Map<String, String> properties) {
+
+        // Validate the minimum required properties are set.
+        if (!properties.containsKey(CatalogProperties.CATALOG_IMPL)) {
```

Comment:
I wonder if we should have helper that goes from `type` -> `catalog-impl` to match how pyiceberg is configured.

```python
class CatalogType(Enum):
    REST = "rest"
    HIVE = "hive"
    GLUE = "glue"
    DYNAMODB = "dynamodb"
    SQL = "sql"
```

Reply:
Actually, I see that the Java API accepts this `type` parameter as well in the `buildIcebergCatalog()` function. I think this should be the preferred method to specify database type so I'll update the docs accordingly.